src/main/scala/is/hail/expr/ir/functions/GenotypeFunctions.scala

```diff
@@ -0,0 +1,40 @@
+package is.hail.expr.ir.functions
+
+import is.hail.annotations.Region
+import is.hail.asm4s._
+import is.hail.expr.types._
+import is.hail.utils._
+
+object GenotypeFunctions extends RegistryFunctions {
+
+  def registerAll() {
+    registerCode("gqFromPL", TArray(tv("N", _.isInstanceOf[TInt32])), TInt32()) { (mb, pl: Code[Long]) =>
+      val region = mb.getArg[Region](1).load()
+      val tPL = TArray(tv("N").t)
```

Comment:
Using .t inside the body here is going to be a problem, because this code is called WAY after the method lookup happens when the types are unified and bound.  We solved this in the old function registry by having a "dependent" version that had a chance to capture the value of the bindings.  Note, in that case, if you're unifying multiple methods, you still need to re-unify because a later, failed unification may have overwritten a binding.

Reply:
The types get re-unified inside emit.

Reply:
So it does!  You're one step ahead as usual.